### PR TITLE
Add `OllamaLLM`

### PR DESCRIPTION
**Pull request by davidberenstein1957**

### CODE REVIEW

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:39]: IMO we shouldn't set `model` with a default

```suggestion
    model: str
```

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:47]:
```suggestion
        """Loads the `AsyncClient` to use Ollama async API."""
```

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:48]: Can the Ollama APIs be protected with an `api_key`? If so, we should include it, I'll create a PR today to include that for Mistral and OpenAI so feel free to reuse (also to be able to provide the API keys in runtime).

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:66]: Here the docstring should contain the description for each of the arguments above i.e. the whole docstring with args, raises and returns.

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:62]: We won't benefit from streaming in any case, so we should disable by default and exclude it from the `args` IMO

```suggestion
```

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:70]: Related to the `stream` arg exclusion mentioned above.

```suggestion
            stream=False,
```

File: src/distilabel/llm/ollama.py
Comment (alvarobartt) [inline src/distilabel/llm/ollama.py:67]: Does the Ollama client allow something like `n` for the OpenAI client to control the number of generations per input? Also the current args look limited to me, is there any way we can include more sampling/generation args?

File: tests/unit/llm/test_ollama.py
Comment (alvarobartt) [inline tests/unit/llm/test_ollama.py:28]:
```suggestion
    def test_ollama_llm(self) -> None:
```

File: src/distilabel/llm/ollama.py
Comment (davidberenstein1957) [reply src/distilabel/llm/ollama.py:48]: It cannot

File: src/distilabel/llm/ollama.py
Comment (davidberenstein1957) [reply src/distilabel/llm/ollama.py:66]: The current args are limited because they are all captured in the `options` which is the normal behavior for `ollama`. Additionally, a lot of these variables are defined in the model card.

Comment (davidberenstein1957) [reply src/distilabel/llm/ollama.py:66]: And it does not allow for `n` per input.

File: src/distilabel/llm/ollama.py
Comment (gabrielmbmb) [inline src/distilabel/llm/ollama.py:88]: Instead of doing this for-loop, I would override the `generate` method like here: https://github.com/argilla-io/distilabel/blob/901fe7ddaf1a284521c37759a2d052a8f5083963/src/distilabel/llm/mistral.py#L141-L164

**Review by gabrielmbmb — CHANGES_REQUESTED** [review]
LGTM! If we can remove the for-loop before merging that would be nice

**Pull request merged**
